packages/entity/src/AuthorizationResultBasedEntityMutator.ts

```diff
@@ -517,45 +518,33 @@ export class AuthorizationResultBasedUpdateMutator<
       EntityMetricsMutationType.UPDATE,
       this.entityClass.name,
       this.queryContext,
-    )(this.updateInTransactionAsync(false, null));
+    )(this.updateInTransactionAsync(false));
   }
 
-  private async updateInTransactionAsync(
-    skipDatabaseUpdate: true,
-    cascadingDeleteCause: EntityCascadingDeletionInfo,
-  ): Promise<Result<TEntity>>;
-  private async updateInTransactionAsync(
-    skipDatabaseUpdate: false,
-    cascadingDeleteCause: EntityCascadingDeletionInfo | null,
-  ): Promise<Result<TEntity>>;
-  private async updateInTransactionAsync(
-    skipDatabaseUpdate: boolean,
-    cascadingDeleteCause: EntityCascadingDeletionInfo | null,
-  ): Promise<Result<TEntity>> {
```

Comment:
going to investigate making `skipDatabaseUpdate` a constructor method as well to not lose this type precision.

Reply:
quick update: doesn't make as much sense as a constructor param. going to leave it as-is for now. loss of type precision is okay due to unit test coverage.